components/proxy/src/main/java/com/hotels/styx/infrastructure/configuration/yaml/JsonNodeConfig.kt

```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright (C) 2013-2022 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.infrastructure.configuration.yaml
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.NullNode
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import com.hotels.styx.api.configuration.Configuration
+import com.hotels.styx.api.configuration.ConversionException
+import com.hotels.styx.infrastructure.configuration.json.ObjectMappers
+import java.io.IOException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.Objects
+import java.util.Optional
+
+/**
+ * Configuration from a jackson JSON node. This class does not do any JSON parsing - it is just a bridge between the
+ * jackson API and Styx API.
+ */
+class JsonNodeConfig private constructor(rootNode: JsonNode?, mapper: ObjectMapper) : Configuration {
+    private val mapper: ObjectMapper
+    private val rootNode: JsonNode
+
+    /**
+     * Construct an instance from a JSON node.
+     *
+     * @param rootNode a JSON node
+     */
+    constructor(rootNode: JsonNode?) : this(rootNode, YAML_MAPPER)
+
+    /**
+     * Construct an instance from a JSON node.
+     *
+     * @param rootNode a JSON node
+     * @param mapper   mapper to convert JSON into objects
+     */
+    init {
+        this.rootNode = rootNode ?: NullNode.getInstance()
+        this.mapper = ObjectMappers.addStyxMixins(Objects.requireNonNull(mapper))
```

Comment:
There is a Kotlin function that can be used here

```suggestion
        this.mapper = ObjectMappers.addStyxMixins(requireNotNull(mapper))
```

Reply:
fixed